File: code/Python/Day8_BuySellStock_II.py
#JUST ONE TRANSACTION ALLOWED HERE

class Solution:
    def maximumProfit(self, prices):
        # code here
        min_price=float("inf")
        max_profit=0
        
        for price in prices:
            if price<min_price:
                min_price=price
            else:
                max_profit = max(max_profit,price-min_price)
        
        return max_profit

File: code/Cpp/Day8_BuySellStock_II.cpp
//Constraint: JUST ONE TRANSACTION TO MAXIMISE PROFIT

class Solution {
  public:
    int maximumProfit(vector<int> &prices) {
    
        int min_price=prices[0];
        int profit=0;
        
        for(int i:prices){
            if(i<min_price) min_price=i;
            
            else{
                profit= max(profit,i-min_price)   ;     
            }
        }
        return profit;
    }
};
